### DISCUSSION TITLE
Choose what name to display in repositories.

### DISCUSSION THREAD

**Discussion by ArkenStorm** [community/community#10840] [Other Feature Feedback, Questions, & Ideas]
Per the discussion and suggestion [here](https://github.community/t/show-first-last-name-instead-of-account-name/2714/20), it seems like a lot of workplaces have developers use their personal github accounts in their organization, but there is no way to have github display the "Name" part of your profile by default. 
This would be a fantastic addition and helpful in many organizations.
Perhaps if there were a per-repository or per-organization "preference" for what name to display by default?

**Comment by Rabattkarte** [comment]
I feel exactly like @xaved88 in the og discussion:

> I don’t want to force my professional workspaces to look at my personal non-descript handle, nor do I want to “create a new account” for each project that I’m a part of.

**Comment by rainabba** [comment]
I'm part of a large org that's migrating from BitBucket. We have users creating entirely new accounts on GitHub JUST because of this concern, something easily addressed in the web UI and should be adjustable by org and user (org taking precendence.)

We aren't talking about global or repo config (though that has to be addressed too), but on the GitHub web UI, there's no good reason this isn't already resolved if the codebase is healthy and GitHub has concern about its users very real-world needs.

**Reply by RalphHightower** [reply]
I, for one, do not want share my personal GitHub account with my employer. What happens when you leave that employer? Will the employer disable that GitHub account since they are no longer an employee? It makes sense for an employer to disable accounts of former employees.

**Reply by Ulrar** [reply]
You don't, they add your account to their organization but the account stays yours, they don't get any access to it.
When you leave they'll take you out of their organization, so you'll lose access to company resources but that's it

**Reply by RalphHightower** [reply]
Thank you. Retaining my personal account is not an issue since I'm now retired.

**Reply by rainabba** [reply]
Each secure org I've joined had me take an extra step to authorize keys that will have access to their org/repos. Those keys would get disabled when you leave, but that's the only long-term impact and one that makes sense to me. This thread is more about how your co-workers will "discover" you on GitHub without already knowing your github username.

Example:
![image](https://user-images.githubusercontent.com/553630/232492680-074ed5bd-909b-4091-aefc-b9603a8a2ef5.png)

**Comment by philjb** [comment]
I'm also in favor of a new feature allowing organization specific profile names (slightly different than the original idea here).

For our company, we don't want to force employees into creating a handle or adding a profile name that is their real name (or even close to it) because we want our employees to be empowered to make the public internet identity choice that is right for him or her. Since the handles may be opaque to keep a person private, a great feature would be to have an "organization name" that is a name that shows like the GitHub profile name but is only visible within the organization and takes preference over the standard GitHub profile name at https://github.com/settings/profile.

**Reply by kapcod** [reply]
In our company we asked all developers to put their github nick as slack field so that, at least though slack, we can identify who is who. Another requirement is to have work email be defined and always used for commits on company repos. Of course having actual name, visible and searchable, would be better.

**Reply by AatifAli666** [reply]
jaxjsxjsjxj

**Comment by mattgrogan** [comment]
This is a major problem - I can't audit usernames when they are "us8yu23wfs" or "new-banana-popcorn"!

**Reply by robbespo00** [reply]
Agree

**Reply by mr-jolly** [reply]
💯

It is incredibly difficult to audit, and I also don't want my unprofessional username to be displayed to everyone in my organization. Also, creating a separate account does not work well because I can not have two accounts associated with my SSO. As I am an admin in my organization, I want to guarantee than permissions have switched over to my new account because detaching my original account. Ideally, I would like to retain both accounts.

GitHub seems to cause me more issues than it solves lately. Pull requests are very poorly implemented. Sometimes I wonder if the GitHub developers actually use their own product.

**Comment by Cubox** [comment]
This is really annoying. I don't want to be showed as "Cubox" on all my company's internal repos, but as my full name. But keep using a single account, and the "Cubox" name for non-company repos

**Comment by skyem** [comment]
We need this at our large company. GitHub's auto-generated changelog in reviews shows the users that closed prs by username, but our company usernames are nonsensical letters and numbers in GitHub Enterprise Cloud, so it's not helpful.

**Comment by DradeAW** [comment]
+1 for this feature!

**Comment by evser** [comment]
+1

**Comment by Lakmus85** [comment]
Agree completely. In organizations it's not really productive to read "K43234 has approved this PR"

**Reply by JamieTaylor94** [reply]
+1

**Reply by robbespo00** [reply]
right

**Reply by ritajie** [reply]
100% agree

**Comment by frozenprakash** [comment]
MUST HAVE FEATURE FOR DEVELOPERS WORKING ON MULTIPLE ORGANIZATION AS WELL AS HAVING COMMON GITHUB FOR PERSONAL AND PROFESSIONAL PROJECTS.

We have around 10k+ developers in our organization and we can't ask every single one to create new duplicate git id just to use for organization.

Even GitHub suggest the same to use same id for personal and professional use to avoid duplication and to capture the stats.
https://github.community/t/using-one-account-for-all-your-projects/10197

But username will not be readable or identifiable with a person in team, so proper feature should allow user to show their full name by default in all GitHub repos and activities.

**Comment by ghost** [comment]
+1

**Reply by Kayleexx** [reply]
fr

**Comment by PetoriousBIG** [comment]
+1. I love my username, but I imagine most people within my organization would rather see my actual name.

**Comment by MolotovCherry** [comment]
Surprised this isn't an option yet. It creates more work for the users just to have to create new accounts with a certain username just so it can display correctly. And if your username is already taken, you're out of luck

**Comment by Conpago-Jayden** [comment]
+1

**Comment by nhooman** [comment]
+1

Migrated the code from GitLab to GitHub and this is a major issue for us. Please fix.

**Comment by jfz** [comment]
+1

**Comment by EdwinFairchild** [comment]
+1

**Comment by mejia-b** [comment]
+1

**Comment by aedanbricker** [comment]
Yeah it would be great to be able to have a username and or just name that I could display for specific organizations. (And if it was a name, just have it display name instead of username in commits and stuff, etc)

This would allow people to stay anonymous on GitHub to the public (so not putting there name in their public profile) but then also still display their real name when working with people that know them in real life or a business setting or whatnot.

Also, as others have stated, this makes the whole environment and interaction look much more professional, which is always a plus.

Honestly, I’m surprised this isn’t already a feature that GitHub offers as I feel like it would be a relatively simple thing to implement that would go a long way.

**Comment by ThinAaronGB** [comment]
Hi! First, I'd like to say that your emulator is so cool! I've been using it since May of 2022 and I can't even tell you how impressive this emu is! The quality of the games in 4k plus the Vulkan system you operate in your emulator is out of this world! Thank you for all this really. I stopped using Duckstation for some time though, and I ended up deleting it altogether, then around September, I decided I wanted to reinstall it for the second time so off I went, however, this time some titles that I love, one of them being Silent Hill 1, began to crash while playing. It automatically took me out of the game with no apparent reason, something that had never happened back then in May, (first time I tried Duckstation on my MacBook Air M1) plus some other issues like in some video clips suddenly everything freezes out. I honestly would like to continue to use this emulator as in my opinion it's the best PS1 emulator under the sun and I love the design and the aesthetics of the games thanks to Duckstation and thanks to all of you who have made this possible. I hope you can work this/these issue(s) out. I don't know why this happens but I am 100 percent sure it did not happen before. Especially with this one game Silent Hill among some other titles too. Thank you for your time! Please let me know how I can contribute to this project so that I can help you somehow.
Kindest regards! Aaron.

**Comment by alimttth** [comment]
Hello, the best tutorials are on YouTube, go there
Have a good fast

**Comment by iTrooz** [comment]
+1

**Comment by tellSlater** [comment]
Can this happen one day soon? +1

**Comment by Zinnavoy** [comment]
1 year already. +1

**Comment by OrchidForever** [comment]
+1 

I don't want to change my name publicly but only for the org they want my legal name which isn't what I go by...

**Comment by gregorpesek** [comment]
Been using GitHub for god knows how many years just to leak off of. Yesterday I contributed my first discussion and today I'm here.. 
Why can't I change my x years old username or just have custom name displayed?

It's bonkers to me seeing GitHub not allow such customization, when everything that's being built here is about customization..

Edit: Alright, my bad. It is possible to change username. Good luck to everyone else!

**Comment by kodfun** [comment]
+1

**Comment by QAscoop** [comment]
+1

**Comment by Sarah4x** [comment]
+1

**Comment by Aed3r** [comment]
+1

**Comment by sdrothrock** [comment]
+1

**Comment by pentago** [comment]
Me too +1

**Comment by jarith** [comment]
+1

**Comment by Hassanlasheenn** [comment]
+1

**Comment by Oisangui** [comment]
+1

**Comment by nicolas-s** [comment]
+1

**Comment by bjorntore** [comment]
+1

**Comment by frozenprakash** [comment]
Hi,
Surprised to see now it's showing hr!

I haven't switched on Google fit, seems to be pulling hr from pixel watch
automatically and updating in Google fit after couple of hours!!

❤️

**Reply by Ulrar** [reply]
:wat:

**Comment by Kahera** [comment]
+1

**Comment by khaamidab** [comment]
+1

**Comment by milek** [comment]
+1

**Comment by l-kldzj** [comment]
+1

**Comment by yuhsinshih** [comment]
+1, surprised this is not an option. My organization's username is unreadable a mix of random number + letter (e.g. a2s34be)

**Reply by millsks** [reply]
Same here.  Yes we can hover and see who it is, but that got old real quick!

**Comment by jongoestoeleven** [comment]
+10

**Comment by pawelkedra-intive** [comment]
+1

**Comment by millsks** [comment]
+1

**Comment by mjarends** [comment]
This is a major issue for my company after having recently migrated from GitLab to GitHub. We would prefer to have the user's name displayed (i.e. Jane Doe) vs our internal company ID which does not have any meaning.

**Comment by algo7** [comment]
+1

**Comment by biosonic** [comment]
+1

**Comment by DefaultO** [comment]
+1

**Comment by LukBed** [comment]
+1

**Comment by chris-shelton** [comment]
+1 

We just migrated onto Github Enterprise from GitLab... it is mind boggling that we cannot display the display names of our users AND within the org, be able to set what the display name is. Just for displaying within the Org, outside, we don't care. 

But this is ridiculous.

**Comment by michael-letcher** [comment]
Agreed, ridiculous this isn't an option.

**Comment by Ray-Eldath** [comment]
PLEASE MAKE THIS INTO A FEATURE. It's so important for us who are using their personal profile in the company... Just add an option to show "Organizational Name" to other people in the same organization, and default ("public") name to everyone else and to the unlogined users.

**Comment by cyberfury10** [comment]
+1

**Comment by Spixmacaw** [comment]
repositritories name is keep according topic and your decision mindset

**Comment by kralle333** [comment]
+1

**Comment by Fregnen** [comment]
+1

**Comment by cyberfury10** [comment]
Hi, As a workaround, I have created an extension **WebEdit Pro -** ([Chrome extension](https://chrome.google.com/webstore/detail/webedit-pro/poacbppnkkaljgcfcgmabigfjhbkfdle), [Edge addon](https://microsoftedge.microsoft.com/addons/detail/webedit-pro/iglmlbhnbljiecdengccffmjlcgbjnei)) that can replace the user name with the desired full name.

Let's say usernames are random alphanumeric strings

K43234, K43235 & so on. Imagine there are 100 users under that repository with auto-generated alphanumeric IDs and whenever there is something happened in the repo. we will see

"K43234 committed a new change"
"K43235 raised a new PR"

With the above chrome extension & mapping as follows,

K43234 -> "John"
K43235 -> "Bruce"

The result would be:
"John committed a new change"
"Bruce raised a new PR"

I added capabilities to perform bulk edit (like in Postman), it is easy to add hundreds of username mapping at once

One can get the username and full name from the **_people_**  or **_Teams_** tab,

<img width="1657" alt="Screenshot 2023-08-24 at 2 10 11 PM" src="https://github.com/community/community/assets/45261873/73fcd2db-41e9-4a66-b646-128f37df744f">

and use notepad, regex to format the username and full name as expected by the extension. It is just one time work though.

here is the repo [link](https://github.com/cyberfury10/web-edit-pro) if you are interested to see the code & video [link ](https://www.youtube.com/watch?v=5j3OCeYzKFY)on how to use

**Reply by AnakinPt** [reply]
I suffer for that pain. I need to hover the username so I know who did a commit or something. Our usernames are of that style "K65GT4"

**Comment by HeroCC** [comment]
Not sure if it's been mentioned before, but this seems close: https://docs.github.com/en/organizations/managing-organization-settings/managing-the-display-of-member-names-in-your-organization

It displays the profile name next to the username in several places. Org-level configuration

**Reply by cyberfury10** [reply]
Yes, It does help to some extent, but doesn't feel convenient. 
When viewing commits, checking the list of reviewers, and in PR tabs, it's necessary to hover over usernames to view their full names

**Reply by NoSugarCoffee** [reply]
It's only avaliable for private repos

**Comment by jacob-lassen** [comment]
+1

**Comment by x3igh7** [comment]
+1

**Comment by zahid60** [comment]
Agree completely.

**Comment by rocifier** [comment]
We struggle daily with this as we have people from other teams at work contributing to our code base and we don't know who they are when we look through pull requests

**Comment by Uliana200407** [comment]
GitHub often encourages users to submit feature requests or suggestions through their official channels. You can submit your feature request directly through the GitHub feedback form, which is accessible through your GitHub account. #10285

**Comment by eml-c2935** [comment]
3000% agree on display name versus username. For organization that using "code" for their username, this is kind of weirdest thing on earth.

**Comment by ploki** [comment]
it is a most needed feature

**Comment by get951753** [comment]
+1

**Comment by AK3030** [comment]
+1

**Comment by fydo** [comment]
Hello, as mentioned by others in this thread, it would be appreciated to be able to customize your personal "display name" on a per-organization level. Thank you!

**Comment by AnonymousSB** [comment]
+1 on this needing to be addressed. Just had a very hectic stand-up as scrum master where I spent about 5 minutes trying to figure out who "Big-Daddy-1" is on the team.

**Reply by millsks** [reply]
Lover of BioShock or Adam Sandler movies? =D
![image](https://github.com/community/community/assets/35641675/86cf4b8b-27cd-460f-a4ae-66d0520ad506)

**Reply by cyberfury10** [reply]
I am using this text replacing extension that I created **WebEdit Pro** - ([Chrome extension](https://chrome.google.com/webstore/detail/webedit-pro/poacbppnkkaljgcfcgmabigfjhbkfdle), [Edge addon](https://microsoftedge.microsoft.com/addons/detail/webedit-pro/iglmlbhnbljiecdengccffmjlcgbjnei)) It solves the issue.
here is the repo [link](https://github.com/cyberfury10/web-edit-pro) if you are interested to see the code & video [link ](https://www.youtube.com/watch?v=5j3OCeYzKFY) on how to use it to solve this github issue

One can get the username and full name from the **_people_**  or **_Teams_** tab, of the repo page

<img width="1657" alt="Screenshot 2023-08-24 at 2 10 11 PM" src="https://github.com/community/community/assets/45261873/73fcd2db-41e9-4a66-b646-128f37df744f">

and use notepad, regex to format the username and full name as expected by the extension. It is just one time work.

**Comment by bab2683** [comment]
+1

**Comment by O5ten** [comment]
I can only second what is being said on this topic. It is annoying that we need to resolve this issue with the help of a browser extension.

**Comment by hardcorewayx** [comment]
+1

**Comment by PolkadotDom** [comment]
+1

**Comment by Blind-Striker** [comment]
+1

**Comment by ar646** [comment]
+1

**Comment by eduardocelebratix** [comment]
+1

**Comment by stefan-gavrilas** [comment]
+1

**Comment by query-csharp** [comment]
+1

**Comment by l0ki000** [comment]
+1

**Comment by littlekath25** [comment]
+1

**Comment by Waterstraal** [comment]
+1

**Comment by MikALito** [comment]
+1 There should be an option to set names per organization.

**Comment by whiteeat** [comment]
+1

**Comment by jotadrilo** [comment]
+1

**Comment by nikican** [comment]
+1

**Comment by fleps** [comment]
GitHub, respectfully, it's 2024.

I'm sure your own Copilot can solve this in like a few minutes.

It's 2024 guys.

**Reply by airtonix** [reply]
@fleps are you saying that it's the current year? you jest surely.

**Comment by galecore** [comment]
+1

**Comment by m20b27** [comment]
+1

**Comment by spacebluebox** [comment]
+1

**Comment by obalais** [comment]
+1

**Comment by Kasp42** [comment]
+1

**Comment by Varrah** [comment]
Please refrain from posting +1, instead, ensure you've pushed the "Top" arrow button on the original post above.

**Comment by Varrah** [comment]
Also, please spread the word between your colleagues in all your organizations, so that they come and push the top button as well. Let's lift it up to the GitHub devs attention. Any other ideas appreciated.

**Comment by Bohousek007** [comment]
Nice! That would be amazing.

**Comment by opetany93** [comment]
+1

**Comment by maddendd-uob** [comment]
Up voted. We are in the process of migrating from BitBucket to GitHub and this is a real quality of life issue. Unrecognisable usernames everywhere rather than real names.

**Reply by O5ten** [reply]
> Up voted. We are in the process of migrating from BitBucket to GitHub and this is a real quality of life issue. Unrecognisable usernames everywhere rather than real names.

It was the exact same thing for us and took some time to get used to.

**Reply by cyberfury10** [reply]
There is browser extension which solves this issue. Check my [comment](https://github.com/orgs/community/discussions/10840#discussioncomment-6810901) above for more info.

**Comment by Norbz** [comment]
+1

**Comment by ThisIsStupidGitHub** [comment]
I see They give zero f***s about the display name after 2.5 YEARS! I mean wtf is even the Point of having a "Public Profile" name if it isn't used ANYWHERE that matters?

**Comment by zzJinux** [comment]
Related: https://github.com/orgs/community/discussions/4168

**Comment by TeTiRoss** [comment]
\+

**Comment by joao-squillace-razr** [comment]
+1

**Comment by RoyTinker** [comment]
This won't be fixed because it's an upsell opportunity for GitHub. They want organizations to use EMU (enterprise managed users) which costs more.

**Reply by AnakinPt** [reply]
This is not related. When I hover the username I see the name of the user. I just want to see the person name instead of an identifier

**Comment by gitclonehama** [comment]
2025, still wondering why we hover to reveal name with the username as default view in discussions when it should logically be the other way around.

 I want to see "Hama Pashazadeh" as the default view and "hamacodes" upon hovering, not the backwards state it is in currently. 
 
 GitHub, pls fix.

**Reply by cyberfury10** [reply]
There is browser extension which solves this issue. Check my [comment](https://github.com/orgs/community/discussions/10840#discussioncomment-6810901) above for more info.

**Comment by colindean** [comment]
My org is considering GHEC and must use Enterprise Managed Users, resulting in unintelligible, undifferentiable, multi-character usernames akin to an eight-character hash. This will impede communication, even with autocomplete suggestions usable through regular names. Hover cards aren't a fast solution, either.

Has anyone developed a browser extension that can replace a username with the user's profile name on each page load?

**Reply by cyberfury10** [reply]
Yes, I have developed browser extension to solve this. Check my [comment](https://github.com/orgs/community/discussions/10840#discussioncomment-6810901) above for more info.

**Reply by AnakinPt** [reply]
I confirm. I had to use this plugin and works perfectly.
It's dumb that we can't pick what string to render... Just stupid.

**Reply by colindean** [reply]
Thanks for the link. I'll take a look.

I'm looking for something specifically scoped to GitHub.com so I don't have to think about permissions much.

**Reply by mr-jolly** [reply]
This still means that you have to have every dev install a browser extension which to me seems like a hassle. Not to mention different companies lock things down differently.

**Reply by Spunkie** [reply]
> I'm looking for something specifically scoped to GitHub.com so I don't have to think about permissions much.

@colindean Scoped to just GH or not, you can be sure using an extension like this is a quick way to get fired. Especially for a GHEC or any other high security biz.

**Reply by colindean** [reply]
Thanks for the concern. It is warranted in general– no shade toward you, @cyberfury10! If I continue with a solution involving custom software, I could build upon that great extension, but limit its permissions and mix in some GitHub API usage so that little to no configuration is required.

**Comment by koshera** [comment]
Still can't believe this is not an option yet...

**Comment by Mehmango** [comment]
+1

**Comment by pgcd** [comment]
+1

**Comment by LindseyCSA** [comment]
This has been unanswered since 2022.  How does this get put in front of GitHub developers for consideration?  I'm just in the process of setting up our company GitHub organization and this is already an issue.  We have outside contractors as well, who also have visibility to GitHub usernames that are at worst potentially offensive, and at best simply not usable to identify who a particular member actually is without too much effort.

**Comment by mattseymour** [comment]
This seems like a no brainer option in github for enterprises, why there is still no comment or movement on this in over 3 years is not great.

**Comment by Yokomoko** [comment]
Insanity that Github still hasn't implemented this for Organizations or Enterprises.

**Comment by opetany93** [comment]
+1

**Comment by jeremieamstutz** [comment]
I fixed it with a chrome extension: https://github.com/jeremieamstutz/github-real-names

**Reply by vgarcia007** [reply]
This really solves the problem for me. Using it makes life on GitHub so much easier. Thank you!

**Reply by jeremieamstutz** [reply]
Thanks :)

**Reply by ThePants999** [reply]
Oohh, this is superb, thanks.

**Comment by clutchwave** [comment]
Adding my +1 for this. I have my username set to my original business name, been using this for over 13 years... but if I want to use something like Giscus for comments, none of my readers will know who I am.

This should be a very very simple thing to set. Not sure what the holdup is. Just have it print my Name as the Display name....

**Comment by mattseymour** [comment]
4 years on and still no true solution. Thanks @github.

We are a team of 40 and keeping track of users is becoming a p.i.t.a. I dont know how larger orgs are managing to do user management. 


Please add support for users real names enforcable over an org.

**Comment by PressXtoChris** [comment]
GitHub has added the ability to display profile names alongside usernames in GitHub Enterprise:

https://github.blog/changelog/2026-02-13-you-can-now-show-profile-names-alongside-user-handles/

https://docs.github.com/en/enterprise-cloud@latest/admin/managing-your-enterprise-account/configuring-user-display-names-for-your-enterprise

**Reply by Blue101black** [reply]
Your organization has to be part of an enterprise still... that's annoying.

**Reply by ArkenStorm** [reply]
Holy moly
Thank you for pointing this out @PressXtoChris !
<img width="565" height="240" alt="about_time" src="https://github.com/user-attachments/assets/8f63bf91-8b91-4f49-a6b9-e52880c7f016" />

**Comment by tkaepp** [comment]
Love the change to enable it on the enterprise and the org! It helps our org and was one of the most basic features that was missing after moving to github. 

An improvment for our enterprise would be that we could configure to select the Display Name from our SSO Provider. Or as an alternative that you can set a display name per enterprise or organisation.

At the moment everybody needs to have the Display Name public on the internet. There are privacy concerns to that.

**Comment by mattseymour** [comment]
+1 on this. It's great to see this feature in the platform but it only
works if users have a display name set. It still doesnt help me
identify users with no display name and obscure account names.

**Comment by airtonix** [comment]
I think we can all give up. 

Microsoft owns github (little g). 

It's a safe bet all outstanding bugs won't be fixed.

**Comment by Ghedeon** [comment]
Can't you ask your copilot to code this feature or something @github? 4 years, jesus christ.

**Comment by yourkarma6788** [comment]
THE THING IS, I HAVE NEVER AUTHORIZED ANY OF THE RESPIRATOROES THAT ARE ACTIVE. THEREFORE, WHI IS VIOLATING MY RIGHTS??

**Comment by lumimario** [comment]
Just making some noise here. This thing is displaying my alias and not my name. This doesn't mean anything in my organisation.

**Comment by MartinsPuspurs** [comment]
Coming any soon?

**Comment by Tryum** [comment]
As we have routing emails, we should be able to have per org aliases... On the web it's ok, but on third party services, I still appear as `tryum` in notifications... that's find by me, but I have to advertise it's me 😅️
